client/components/LearnerViewDirective.js

```diff
@@ -465,8 +467,19 @@ tie.directive('learnerView', [function() {
         };
 
         $scope.navigateToQuestion = function(index) {
+          // Before the questionId is changed, save it for later use.
+          var currentQuestionId = 
+            $scope.questionIds[$scope.currentQuestionIndex];
           $scope.currentQuestionIndex = index;
           var questionId = $scope.questionIds[$scope.currentQuestionIndex];
+          // Whenever a user navigates to a different question,
+          // store their code on the scope.
+          var currentQuestionCode = $scope.code;
```

Comment:
I think you don't need this var; just replace currentQuestionCode with $scope.code in the function call below.